### ISSUE TITLE
provide -nonfree flavor

### ISSUE THREAD

**Issue by yarikoptic** [neurodebian/dockerfiles#5]
@teanon what do you think?

quite frequently I need to work in a container which also points to contrib/non-free components since some quite popular packages are non-free... it is kinda annoying to always require installing vim or running some sed command to enable contrib/non-free.  So I wondered if we could streamline it somehow.  I see possibly
1. proiding neurodebian:release-nonfree (e.g. neurodebian:sid-nonfree) flavors as well which would be identical to free, just with apt source adjusted accordingly.  IMHO it is the preferable way
2. provide a copy of apt .sources file with those nonfree suits enabled but the whole file .disabled, so if needed those could simply be mv/cp'ed over... but that is suboptimal since you need to remember/know all apt sources files to copy/modify

so, do you think going `1.` way would be ok? any specific setup I could aim for, so e.g. -nonfree images could be created on top of neurodebian default (main only)  ones?

thanks in advance for the feedback

**Comment by tianon** [comment]
If `neurodebian`+`nonfree` is something really common, then option 1 definitely sounds reasonably sane to me!

You could even go with simple `Dockerfile`s for it something like:
```dockerfile
FROM neurodebian:stretch
RUN adjust-sources-as-desired....
```

**Comment by tianon** [comment]
For example, see `debian:jessie-backports` (https://github.com/debuerreotype/docker-debian-artifacts/blob/0dcb9a06b2fcf9fdb416dc558db6f7b84cb6ab01/jessie/backports/Dockerfile) and `debian:experimental` (https://github.com/debuerreotype/docker-debian-artifacts/blob/0dcb9a06b2fcf9fdb416dc558db6f7b84cb6ab01/experimental/Dockerfile). :+1:

**Comment by yarikoptic** [comment]
Great, thank you, will do (now stuck at the airport only with the cell phone)

**Comment by yarikoptic** [comment]
@tianon Finally did it: https://github.com/docker-library/official-images/pull/3241  to discover that apparently ubuntu 17.04 release docker image (but not previous seems to me) comes with multiverse enabled. Also, I went for somewhat suboptimal way probably where I just tuned up Dockerfile for -non-free flavor to get one more RUN statement at the end.  With caching, probably should be identical to having dedicated Dockerfile which would start with the "free" version and just have a single RUN command.
